File: 3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        n = len(nums)
        nums.sort()
        res = float("inf")

        l,r = 0,0
        while r < n:
            if nums[l]*k >= nums[r]:
                r+=1
                res = min(l+n-r,res)
            else:
                l+=1
        return res
            

File: 3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.cpp
class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        int n = nums.size();
        sort(nums.begin(),nums.end());

        int res = INT_MAX;

        int l = 0;
        int r = 0;

        while (r < n){
            if (nums[l]*k >= nums[r]){
                r++;
                res = min(res,l+n-r);
            }else{
                l++;
            }
        }
    return res;
        
    }
};